zrml/prediction-markets/src/tests.rs

```diff
@@ -1125,6 +1128,350 @@ fn market_resolve_does_not_hold_liquidity_withdraw() {
+        let balance_free_before_alice = Balances::free_balance(&ALICE);
+        assert_ok!(PredictionMarkets::admin_destroy_market(Origin::signed(SUDO), 0));
+        assert_eq!(Balances::reserved_balance_named(&RESERVE_ID, &ALICE), SENTINEL_AMOUNT);
+        let balance_free_after_alice = Balances::free_balance(&ALICE);
+        assert_eq!(balance_free_before_alice, balance_free_after_alice);
+    });
+}
+
+#[test]
+fn admin_destroy_market_correctly_cleans_up_accounts() {
+    ExtBuilder::default().build().execute_with(|| {
+        let amount = <Runtime as zrml_swaps::Config>::MinLiquidity::get();
+        assert_ok!(PredictionMarkets::create_cpmm_market_and_deploy_assets(
+            Origin::signed(ALICE),
+            ALICE,
+            MarketPeriod::Block(0..42),
+            gen_metadata(50),
+            MarketType::Categorical(3),
+            MarketDisputeMechanism::SimpleDisputes,
+            <Runtime as zrml_swaps::Config>::MinLiquidity::get(),
+            vec![amount, amount, amount],
+            vec![<Runtime as zrml_swaps::Config>::MinWeight::get(); 4],
+        ));
+        let market_id = 0;
+        let pool_id = 0;
+        let pool_account = Swaps::pool_account_id(pool_id);
+        let market_account = PredictionMarkets::market_account(market_id);
+        let alice_balances_before = vec![
+            Currency::free_balance(Asset::CategoricalOutcome(0, 0), &ALICE),
+            Currency::free_balance(Asset::CategoricalOutcome(0, 1), &ALICE),
+            Currency::free_balance(Asset::CategoricalOutcome(0, 2), &ALICE),
+            Currency::free_balance(Asset::Ztg, &ALICE),
+        ];
+        assert_ok!(PredictionMarkets::admin_destroy_market(Origin::signed(SUDO), 0));
+        assert_eq!(Currency::free_balance(Asset::CategoricalOutcome(0, 0), &pool_account), 0);
+        assert_eq!(Currency::free_balance(Asset::CategoricalOutcome(0, 1), &pool_account), 0);
+        assert_eq!(Currency::free_balance(Asset::CategoricalOutcome(0, 2), &pool_account), 0);
+        assert_eq!(Currency::free_balance(Asset::Ztg, &pool_account), 0);
+        assert_eq!(Currency::free_balance(Asset::CategoricalOutcome(0, 0), &market_account), 0);
+        assert_eq!(Currency::free_balance(Asset::CategoricalOutcome(0, 1), &market_account), 0);
+        assert_eq!(Currency::free_balance(Asset::CategoricalOutcome(0, 2), &market_account), 0);
+        assert_eq!(Currency::free_balance(Asset::Ztg, &market_account), 0);
+        assert_eq!(
+            Currency::free_balance(Asset::CategoricalOutcome(0, 0), &ALICE),
+            alice_balances_before[0]
+        );
+        assert_eq!(
+            Currency::free_balance(Asset::CategoricalOutcome(0, 1), &ALICE),
+            alice_balances_before[1]
+        );
+        assert_eq!(
+            Currency::free_balance(Asset::CategoricalOutcome(0, 2), &ALICE),
+            alice_balances_before[2]
+        );
+        assert_eq!(Currency::free_balance(Asset::Ztg, &ALICE), alice_balances_before[3]);
```

Comment:
So Alice looses nothing? I thought the assets get slashed :-O ? This checks that Alice free balance of each asset is the same as before. Am I missing something here?